plugins/scaffolder-backend/src/scaffolder/actions/builtin/ci/githubActionsDispatch.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { InputError } from '@backstage/errors';
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { Octokit } from '@octokit/rest';
+import { createTemplateAction } from '../../createTemplateAction';
+import { Config } from '@backstage/config';
+
+const host = 'github.com';
+
+export function createGithubActionsDispatchAction(options: {
+  integrations: ScmIntegrationRegistry;
+  config: Config;
+}) {
+  const { integrations } = options;
+
+  const credentialsProviders = new Map(
+    integrations.github.list().map(integration => {
+      const provider = GithubCredentialsProvider.create(integration.config);
+      return [integration.config.host, provider];
+    }),
+  );
+
+  return createTemplateAction<{
+    owner: string;
+    repoName: string;
+    workflowId: string;
+    branchOrTagName: string;
+  }>({
+    id: 'ci:github-actions-dispatch',
+    description:
+      'Dispatches a GitHub Action workflow for a given branch or tag',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['owner', 'repoName', 'workflowId', 'branchOrTagName'],
+        properties: {
```

Comment:
Thinking that this should take a `repoUrl` which is the from the `repoUrlPicker` component. Then you can get rid of the `owner` and `repoName` props from here, and when you use `parseRepoUrl` with the input it will give you the `hostname` too, so will work for `ghe` and `github.com`

Reply:
Updated in [`8fb0874` (#6493)](https://github.com/backstage/backstage/pull/6493/commits/8fb0874dd44c1098d0120fb9e93bb26e9995f250#diff-1e7f5015e36aa57277e63731ed4d0c4c12b864a914069c4ad97dba7d4c4701f2R38)